backend/tests/integration_docker/test_propose_change_repository.py

```diff
@@ -0,0 +1,88 @@
+from tests.helpers.fixtures import get_fixtures_dir
+
+CURRENT_DIRECTORY = Path(__file__).parent.resolve()
+
+
+class TestProposeChangeRepository(TestInfrahubDockerClient, SchemaCarPerson):
+    @pytest.fixture(scope="class")
+    def infrahub_version(self) -> str:
+        return "local"
+
+    @pytest.fixture(scope="class")
+    def schema_person_artifact(self, schema_person_base: NodeSchema) -> NodeSchema:
+        person_schema = schema_person_base.model_copy(deep=True)
+        person_schema.inherit_from = [InfrahubKind.ARTIFACTTARGET]
+        return person_schema
+
+    @pytest.fixture(scope="class")
+    def initial_schema(
+        self,
+        schema_car_base: NodeSchema,
+        schema_person_artifact: NodeSchema,
+        schema_manufacturer_base: NodeSchema,
+    ) -> SchemaRoot:
+        return SchemaRoot(
+            version="1.0",
+            nodes=[schema_person_artifact, schema_car_base, schema_manufacturer_base],
+        )
+
+    async def test_load_initial_schema(
+        self, default_branch: str, client: InfrahubClient, initial_schema: SchemaRoot
+    ) -> None:
+        await client.schema.wait_until_converged(branch=default_branch)
+
+        resp = await client.schema.load(
+            schemas=[initial_schema.to_schema_dict()], branch=default_branch, wait_until_converged=True
+        )
+        assert resp.errors == {}
+
+    async def test_load_initial_data(self, client: InfrahubClient, default_branch: str, remote_repos_dir: Path) -> None:
+        data = await self.create_initial_data(client=client, branch=default_branch)
+        persons = data[TESTING_PERSON]
+
+        # Create Group People
+        group_people = await client.create(
+            kind="CoreStandardGroup", name="people", members=[item.id for item in persons]
+        )
+        await group_people.save()
+
+        # Add repositories
+        fixture_dir = get_fixtures_dir()
+        repo_name = "car-dealership"
+        repo_dir = fixture_dir / "repos" / repo_name / "initial__main"
+        repo = GitRepo(name=repo_name, src_directory=repo_dir, dst_directory=remote_repos_dir)
+        await repo.add_to_infrahub(client=client)
+        in_sync = await repo.wait_for_sync_to_complete(client=client)
+        assert in_sync
+
+        repos = await client.all(kind=CoreGenericRepository)
+        assert repos
+
+    async def test_create_propose_change(self, client: InfrahubClient, default_branch: str) -> None:
+        branch = await client.branch.create(branch_name="branch2")
+        john = client.store.get_by_hfid(key=f"{TESTING_PERSON}__John Doe", raise_when_missing=True)
+
+        john_branch = await client.get(kind=TESTING_PERSON, id=john.id, branch=branch.name)
+        john_branch.description.value = "new description"
+        await john_branch.save()
+
+        pc = await client.create(
+            kind=CoreProposedChange, name="pc1", source_branch=branch.name, destination_branch=default_branch
+        )
+        await pc.save()
```

Comment:
should this verify that that proposed change is actually created?

Reply:
all good points, this file is more of an example to get started then a true functional test at some point